### PR TITLE
Update and fix cnc-ddraw

### PR DESCRIPTION
**Pull request by Root-Core**
The current implementation fails, if ddraw.dll is a symlink - like in Proton-GE.

### CODE REVIEW

**Comment by Root-Core** [comment]
> I've never seen any complaints of simply unpacking the archive caused any Linux users issues so I'll assume this would be Proton specific.

Nah. This is why there is the `w_try_cp_dll()` function in first place and it is used for other dlls.

https://github.com/Winetricks/winetricks/blob/f87bf9e6a7c67a06487a1ef710c0d9c548ae6f01/src/winetricks#L748-L751

> If special changes are required for GE-Proton that should really be handled within `umu-protonfixes`,

I don't see a single reason it shouldn't be used here.

**Comment by Root-Core** [comment]
> Yes using w_try_cp_dll() would indeed be valid, is wine on Linux using symlinks more heavily now for prefixes?

I think so. I'm not using wine directly that much though. I know Proton-GE and Lutris are using it.
I also searched for symlinks in Prefixes that use the Proton provides by Steam and they make heavy use out of it.
It makes sense to me, as you don't have to duplicate all files and by replacing a symlink with a file - like in this case - only affects a singular prefix.

> The Linux specific comment was more aimed at the usage of -u but I see you’ve added thumbs up to both my other comments.

I will fix it soon. I have only little experience with BSD / MacOS, so I missed that.

I thought an update might be a good thing, but we could just delete files in order to update them.. if necessary.
Afaik there isn't really a mechanism to update a fix to a newer version what so ever, so just forget what I just said. :)

> Something that might be worth noting is if something is platform specific (not project specific like Proton) it could always be placed behind a platform guard like I’ve done for the Steam verb for macOS.

Good to know, but I don't think that's necessary here. This fix should work on all platforms.

File: src/winetricks [hunk 1]
```diff
@@ -6088,9 +6088,12 @@ load_cnc_ddraw()
     # 2023/11/04 https://github.com/FunkyFr3sh/cnc-ddraw/releases/download/v6.0.0.0/cnc-ddraw.zip
     # 2024/02/03 https://github.com/FunkyFr3sh/cnc-ddraw/releases/download/v6.1.0.0d/cnc-ddraw.zip
     # 2024/02/21 https://github.com/FunkyFr3sh/cnc-ddraw/releases/download/v6.2.0.0/cnc-ddraw.zip
+    # 2024/02/21 https://github.com/FunkyFr3sh/cnc-ddraw/releases/download/v6.3.0.0/cnc-ddraw.zip
 
-    w_download https://github.com/FunkyFr3sh/cnc-ddraw/releases/download/v6.2.0.0/cnc-ddraw.zip e5677ba52c31ffa93421a16edacff0c4d1f03e107aea6fc860861b43e3356119 cnc-ddraw-v6.2.0.0.zip
-    w_try_unzip "${W_SYSTEM32_DLLS}" "${W_CACHE}/${W_PACKAGE}/${file1}"
+    w_download https://github.com/FunkyFr3sh/cnc-ddraw/releases/download/v6.3.0.0/cnc-ddraw.zip c024d0ea42ec2d9708dc0a19342d037de7307c291dc43c01948a7d8f06b4deca ${file1}
```
[the remark this answers is not in the document]
Comment (Root-Core) [reply src/winetricks:6093]: I checked, why it fails and `src/linkcheck.sh` isn't failing, but `tests/shell-checks`.
This is fixed in #2204.

**Comment by Root-Core** [comment]
Updated to v6.8.0.0, could we please merge after #2204 is accepted?

**Comment by Root-Core** [comment]
Updated to v7.0.0.0, could we please merge after https://github.com/Winetricks/winetricks/pull/2204 is accepted?

**Comment by austin987** [comment]
Thanks; I squashed it locally and added a couple fixes (quoting the variable / updating the version in the metadata).

I pushed it directly in https://github.com/Winetricks/winetricks/commit/cede5047101299802a9756f0885e24a5b6f2fbb5

**Pull request closed without merging**